docs/src/architecture/assets.md

```diff
@@ -0,0 +1,61 @@
+
+<p align="center">
+    <img src="../diagrams/architecture/asset/Asset_Issuance.png">
+</p>
+
+### Fungible assets
+A fungible asset is encoded using the amount and the `faucet_id` of the faucet which issued the asset. The amount is guaranteed to be `2^63 - 1` or smaller, the maximum supply for any fungible asset. Examples of fungible assets are ETH and stablecoins, e.g., DAI, USDT, and USDC.
+
+If the `faucet_id` of MATIC were to be `2`, 100 MATIC are encoded as `[100, 0, 0, 2]` - whereas the `0`s in the middle help to quickly distinguish between fungible and non-fungible assets. 
+
+Note: It is impossible to find a collision between two fungible assets issued by different faucets as the `faucet_id` is included in the asset's description. The id is guaranteed to be other for each faucet as per the faucet creation logic.
+
+### Non-fungible assets
+A non-fungible asset is encoded by hashing the asset data into a `Word` and the second element with the `faucet_id` that issued the asset. It looks like `[e0, faucet_id, e2, e3]`. Note that the second element is guaranteed to be non-Zero.
+
+Examples of non-fungible assets are all NFTs, e.g., a DevCon ticket. Here the asset data can be the owner and the ticket number.
+
+It is impossible to find a collision between two non-fungible assets issued by different faucets as the `faucet_id` is included in the description of the non-fungible asset, which is guaranteed to be different per the faucet creation logic. Collision resistance for non-fungible assets issued by the same faucet is `~2^95`.
+
+### Storage
+Assets are stored in accounts and notes. Both have vaults to record the `asset_id` and the respective `amount`. Accounts can keep unlimited assets in a tiered sparse Merkle tree called `asset_tree`. Notes can always only store one type of asset, and their total number is bound to `255`. 
+
+<p align="center">
+    <img src="../diagrams/architecture/asset/Asset_Storage.png">
+</p>
+
+The information on which and how many assets are owned can be private depending on the account's storage mode or the note. This is true for any native asset in Miden. 
+
+## Non-native assets
+Miden is flexible enough to create other types of assets as well. 
+
+For example, developers can replicate Ethereum's ERC20 model, where ownership of fungible assets is recorded in a single account. To transact, users must send a note to that account to change the global hashmap. 
+
+Furthermore, a complete account can be treated as a programmable asset because accounts are transferrable. An account could be a "crypto kitty" with specific attributes and rules, and people can trade these "crypto kitties" by transferring accounts between each other. 
```

Comment:
maybe instead of saying accounts are transferable we could say "account ownership is transferable"